torch/csrc/acc/Module.cpp

```diff
@@ -0,0 +1,205 @@
+  void setDevice(c10::Device device) const override {}
+  void uncheckedSetDevice(c10::Device device) const noexcept override {}
+  c10::Stream getStream(c10::Device) const noexcept override {
+    // no-op
+    return c10::Stream(c10::Stream::DEFAULT, getDevice());
+  }
+
+  c10::Stream getNewStream(c10::Device, int priority = 0) const override {
+    // no-op
+    (void)priority;
+    return c10::Stream(c10::Stream::DEFAULT, getDevice());
+  }
+
+  c10::Stream exchangeStream(c10::Stream) const noexcept override {
+    // no-op
+    return c10::Stream(c10::Stream::DEFAULT, getDevice());
+  }
+  c10::DeviceIndex deviceCount() const noexcept override {
+    return 1;
+  }
+
+  // TODO(qihqi): support Event-related functions
+  void record(
+      void** /*event*/,
+      const c10::Stream& /*stream*/,
+      const c10::DeviceIndex /*device_index*/,
+      const c10::EventFlag /*flag*/) const override {}
+  void block(void* /*event*/, const c10::Stream& /*stream*/) const override {}
+  bool queryEvent(void* /*event*/) const override {
+    return true;
+  }
+  void destroyEvent(void* /*event*/, const c10::DeviceIndex /*device_index*/)
+      const noexcept override {}
+
+  // Stream-related functions
+  bool queryStream(const c10::Stream& /*stream*/) const override {
+    return true;
+  }
+  void synchronizeStream(const c10::Stream& /*stream*/) const override {}
+};
+
+namespace {
+
+static py::object g_hook;
+bool registerPythonPrivateUse1Hook(py::object hook) {
+  if (at::isPrivateUse1HooksRegistered()) {
+    return false;
+  }
+  g_hook = hook;
+  at::RegisterPrivateUse1HooksInterface(
+      hook.cast<PrivateUse1HooksInterface*>());
+  return true;
+}
+
+static py::object g_guard;
+bool registerPythonPrivateUse1DeviceGuard(py::object guard) {
+  if (c10::impl::hasDeviceGuardImpl(c10::DeviceType::PrivateUse1)) {
+    return false;
+  }
+  g_guard = guard;
+  c10::impl::registerDeviceGuard(
+      c10::DeviceType::PrivateUse1,
+      g_guard.cast<c10::impl::DeviceGuardImplInterface*>());
+  return true;
+}
+
+void cleanupPyObj() {
+  g_guard.release();
+  g_hook.release();
+}
+
+at::Tensor createEmptyTensor(
+    const std::vector<int64_t>& shape,
+    c10::ScalarType dtype) {
+  c10::Storage storage{
+      c10::Storage::use_byte_size_t{},
+      0,
+      c10::GetAllocator(c10::kMeta),
+      true,
+  };
+
+  c10::Device device(c10::DeviceType::PrivateUse1, 0);
+  storage.set_data_ptr_noswap(at::DataPtr{nullptr, device});
+  c10::DispatchKeySet key_set({c10::DispatchKey::PrivateUse1});
+  at::Tensor tensor = at::detail::make_tensor<at::TensorImpl>(
+      std::move(storage), key_set, c10::scalarTypeToTypeMeta(dtype));
+
+  std::vector<int64_t> strides(shape.size());
+  int64_t size = 1;
+  for (int64_t i = strides.size() - 1; i >= 0; --i) {
+    strides[i] = size;
+    size *= shape[i];
+  }
+
+  tensor.unsafeGetTensorImpl()->set_sizes_and_strides(shape, strides, 0);
+  return tensor;
+}
+} // namespace
+
+void initModule(PyObject* module) {
```

Comment:
As setup here, all of these will end up on `torch.*` because we have a `from _C import *` in the main `__init__` file.
The simplest way to avoid that is to either define a subfield here like `torch._C._acc` or prepend a `_` to every function and classes here.
Note that you will see an error in CI reflecting this once CI runs.